src/transformers/trainer.py

```diff
@@ -1442,6 +1442,18 @@ def optimizer_hook(param):
                 optimizer_cls = Lomo
 
             optimizer_kwargs.update({"model": model})
+        elif args.optim == OptimizerNames.GROKADAMW:
+            try:
+                from grokadamw import GrokAdamW
+                optimizer_cls = GrokAdamW
+                optimizer_kwargs.update({"alpha_init": float(optim_args.get("alpha_init", 0.98)),
```

Comment:
Similar to LOMO - this should be guarded with a `is_grokadamw` available and tests should be added c.f. #30178

Reply:
Thank you for this feedback - I have implemented it as you requested